### PR TITLE
add com.github.brainstormtrooper.facteur

### PR DESCRIPTION
**Pull request by brainstormtrooper**
### Please confirm your submission meets all the criteria

- [ x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [ x] My pull request follows the instructions at [App Submission][submission].
- [ x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
  - d-bus access for saving config and allowing curl to access the internet
- [ x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [ x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [ x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [ x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission

### CODE REVIEW

File: com.github.brainstormtrooper.facteur.json [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    "runtime-version" : "44",
+    "branch" : "stable",
+    "sdk" : "org.gnome.Sdk",
+    "command" : "com.github.brainstormtrooper.facteur",
+    "finish-args" : [
+        "--share=ipc",
+        "--socket=fallback-x11",
+        "--socket=wayland",
+        "--socket=system-bus",
+        "--socket=session-bus",
```
Comment (hfiguiere) [inline com.github.brainstormtrooper.facteur.json:13]: these dbus permissions are NOT allowed. You need to determine which dbus names you need to access.

File: com.github.brainstormtrooper.facteur.json [hunk 2]
```diff
@@ -0,0 +1,52 @@
+{
+    "app-id" : "com.github.brainstormtrooper.facteur",
```
Comment (hfiguiere) [inline com.github.brainstormtrooper.facteur.json:2]: it should be `io.github` as per the documentation.

File: com.github.brainstormtrooper.facteur.json [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        "--socket=system-bus",
+        "--socket=session-bus",
+        "--socket=ssh-auth",
+        "--device=dri",
+        "--share=network",
+        "--filesystem=home:create",
+        "--filesystem=xdg-run/gvfsd",
+        "--filesystem=xdg-run/dconf",
+        "--filesystem=~/.config/dconf",
+        "--talk-name=ca.desrt.dconf",
```
Comment (hfiguiere) [inline com.github.brainstormtrooper.facteur.json:21]: you shouldn't need this dconf stuff.

File: com.github.brainstormtrooper.facteur.json [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    ],
+    "modules" : [
+        {
+            "name" : "facteur",
+            "builddir" : true,
+            "buildsystem" : "meson",
+            "sources" : [
+                {
+                    "type" : "git",
+                    "url" : "https://github.com/brainstormtrooper/facteur/releases/latest"
```
Comment (hfiguiere) [inline com.github.brainstormtrooper.facteur.json:44]: you need a commit or tag.

File: com.github.brainstormtrooper.facteur.json [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+                }
+            ]
+        }
+    ],
+    "build-options" : {
+        
+    }
```
Comment (hfiguiere) [inline com.github.brainstormtrooper.facteur.json:51]:
```suggestion
    ]
```

**Comment by brainstormtrooper** [comment]
These requests have been taken into account

File: io.github.brainstormtrooper.facteur.json [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    "sdk" : "org.gnome.Sdk",
+    "command" : "io.github.brainstormtrooper.facteur",
+    "finish-args" : [
+        "--share=ipc",
+        "--socket=fallback-x11",
+        "--socket=wayland",
+        "--socket=ssh-auth",
+        "--device=dri",
+        "--share=network",
+        "--filesystem=home:create",
```
Comment (hfiguiere) [inline io.github.brainstormtrooper.facteur.json:14]: why do you need `:create`.

Why do you need unfettered acccess to `home`?

File: io.github.brainstormtrooper.facteur.json [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+    "app-id" : "io.github.brainstormtrooper.facteur",
+    "runtime" : "org.gnome.Platform",
+    "runtime-version" : "44",
+    "sdk" : "org.gnome.Sdk",
+    "command" : "io.github.brainstormtrooper.facteur",
+    "finish-args" : [
+        "--share=ipc",
+        "--socket=fallback-x11",
+        "--socket=wayland",
+        "--socket=ssh-auth",
```
Comment (hfiguiere) [inline io.github.brainstormtrooper.facteur.json:11]: why do you need this socket?

File: io.github.brainstormtrooper.facteur.json [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+        "--device=dri",
+        "--share=network",
+        "--filesystem=home:create",
+        "--filesystem=xdg-run/gvfsd",
+        "--filesystem=xdg-run/dconf",
+        "--talk-name=org.freedesktop.secrets",
+        "--env=DCONF_USER_CONFIG_DIR=.config/dconf"
```
Comment (hfiguiere) [inline io.github.brainstormtrooper.facteur.json:18]: still some dconf things left over. (lines 16 and 18)

File: io.github.brainstormtrooper.facteur.json [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+    "command" : "io.github.brainstormtrooper.facteur",
+    "finish-args" : [
+        "--share=ipc",
+        "--socket=fallback-x11",
+        "--socket=wayland",
+        "--socket=ssh-auth",
+        "--device=dri",
+        "--share=network",
+        "--filesystem=home:create",
+        "--filesystem=xdg-run/gvfsd",
```
Comment (hfiguiere) [inline io.github.brainstormtrooper.facteur.json:15]: if you really need gvfs access, this is insufficient.

if you don't need gvfs, then thise should be removed.

https://docs.flatpak.org/en/latest/sandbox-permissions.html#gvfs-access

File: io.github.brainstormtrooper.facteur.json (hunk 6 above)
Comment (brainstormtrooper) [reply io.github.brainstormtrooper.facteur.json:14]: to save files created in the app

File: io.github.brainstormtrooper.facteur.json (hunk 7 above)
Comment (brainstormtrooper) [reply io.github.brainstormtrooper.facteur.json:11]: ? curl ssl/tls, do I not need it?

File: io.github.brainstormtrooper.facteur.json (hunk 8 above)
Comment (brainstormtrooper) [reply io.github.brainstormtrooper.facteur.json:18]: the app does use dconf to store settings, so don't I need access?

File: io.github.brainstormtrooper.facteur.json (hunk 6 above)
Comment (hfiguiere) [reply io.github.brainstormtrooper.facteur.json:14]: if you use Gtk yu shouldn't have to with `GtkNativeFileDialog`.

File: io.github.brainstormtrooper.facteur.json (hunk 7 above)
Comment (hfiguiere) [reply io.github.brainstormtrooper.facteur.json:11]: no

File: io.github.brainstormtrooper.facteur.json (hunk 8 above)
Comment (hfiguiere) [reply io.github.brainstormtrooper.facteur.json:18]: use use dconf directly?

File: io.github.brainstormtrooper.facteur.json (hunk 6 above)
Comment (brainstormtrooper) [reply io.github.brainstormtrooper.facteur.json]: indeed...

File: io.github.brainstormtrooper.facteur.json (hunk 7 above)
Comment (brainstormtrooper) [reply io.github.brainstormtrooper.facteur.json]: removed

File: io.github.brainstormtrooper.facteur.json (hunk 8 above)
Comment (brainstormtrooper) [reply io.github.brainstormtrooper.facteur.json]: removed

File: io.github.brainstormtrooper.facteur.json (hunk 9 above)
Comment (brainstormtrooper) [reply io.github.brainstormtrooper.facteur.json]: removed

**Comment by hfiguiere** [comment]
bot, build io.github.brainstormtrooper.facteur

**Comment by hfiguiere** [comment]
```
io.github.brainstormtrooper.facteur.desktop: AppData problem: tag-missing : <description> required
```

**Comment by brainstormtrooper** [comment]
can you please retry the build?

**Comment by hfiguiere** [comment]
you can do that yourself.

**Comment by brainstormtrooper** [comment]
bot, build io.github.brainstormtrooper.facteur

**Comment by brainstormtrooper** [comment]
bot, build io.github.brainstormtrooper.facteur

**Comment by brainstormtrooper** [comment]
bot, build io.github.brainstormtrooper.facteur

**Comment by brainstormtrooper** [comment]
bot, build io.github.brainstormtrooper.facteur

File: io.github.brainstormtrooper.facteur.json [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+    "modules" : [
+        {
+            "name" : "facteur",
+            "builddir" : true,
+            "buildsystem" : "meson",
+            "sources" : [
+                {
+                    "type" : "git",
+                    "url" : "https://github.com/brainstormtrooper/facteur",
+                    "tag": "0.13.3"
```
Comment (hfiguiere) [inline io.github.brainstormtrooper.facteur.json:35]: please also add the commit that match the tag.

File: io.github.brainstormtrooper.facteur.json [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+    "runtime-version" : "44",
+    "sdk" : "org.gnome.Sdk",
+    "command" : "io.github.brainstormtrooper.facteur",
+    "finish-args" : [
+        "--share=ipc",
+        "--socket=fallback-x11",
+        "--socket=wayland",
+        "--device=dri",
+        "--share=network",
+        "--talk-name=org.freedesktop.secrets"
```
Comment (hfiguiere) [inline io.github.brainstormtrooper.facteur.json:13]: what is this needed for?

Comment (brainstormtrooper) [reply io.github.brainstormtrooper.facteur.json:13]: saving passwords

**Comment by barthalion** [comment]
bot, build io.github.brainstormtrooper.facteur

**Pull request closed without merging**